### ISSUE TITLE
Service Registration and Health Checks flapping when task name starts with another task name

### ISSUE THREAD

**Issue by Ashald** [hashicorp/nomad#2294]
If filing a bug please include the following:

### Nomad version
`Nomad v0.5.0`

### Operating system and Environment details
CentOS 7

### Issue
When job created as described in `Reproduction steps`, the task "that starts with other task's name" if "flaky" in Consul. it's service definition being "registered"/"deregistered" every few seconds and health-checks failing to execute (we tried script and HTTP checks).

### Reproduction steps

Create a job with a group and 2 tasks called `foo` and `foobar`.

### Nomad Client logs (if appropriate)
Executor log
```
017/02/07 17:49:46 [DEBUG] consul.syncer: check "a48e3df662d3e656d9fd5de8d7202c27ff7ee150" failed, disabling Consul checks until until next successful sync: Unexpected response code: 500 (CheckID "a48e3df662d3e656d9fd5de8d7202c27ff7ee150" does not have associated TTL)
```

**Comment by schmichael** [comment]
Pushed a branch with new test reproducing the bug you reported. I'm working on a significant refactor of our consul interaction code that will fix this, but it looks like that refactoring won't be done in time for 0.5.5 I'll try to get a patch in.

**Comment by Ashald** [comment]
Thanks for quick reply! We've used a quick and obvious workaround of changing the task name for now (so we're not in a hurry here). :) But would be nice to have that fixed so looking forward to your refactoring. Would you mind mention/ping me once you going to merge those changes? Thanks

**Comment by schmichael** [comment]
@Ashald I'll definitely close this ticket when the fix lands whether it's in a patch or as part of the larger refactoring!

**Comment by Ashald** [comment]
Thanks!

**Comment by schmichael** [comment]
Fixed on master by #2467
